substra/sdk/client.py

```diff
@@ -949,3 +951,111 @@ def organization_info(self) -> models.OrganizationInfo:
     def cancel_compute_plan(self, key: str) -> None:
         """Cancel execution of compute plan. Nothing is returned by this method"""
         self._backend.cancel_compute_plan(key)
+
+    @logit
+    def wait_compute_plan(
+        self, key: str, *, timeout: Optional[float] = None, polling_period: float = 2.0, raises: bool = True
+    ) -> models.ComputePlan:
+        """Block the execution until the compute plan finishes.
+
+        It is considered finished when the status is done, failed or cancelled.
+        If a timeout is defined, will raises a `FutureTimeoutError` once it reached it.
+
+        Args:
+            key (str): the key of the compute plan to wait
+            timeout (Optional[float]): maximum time to wait. If set to None, will hang until completion.
+            polling_period (float): time to wait between to checks, in seconds. Default to 2.0.
+            raises (bool): wether or not to raises exception if the execution fails. Default to True.
+
+        Returns:
+            models.ComputePlan: the compute plan after completion
+        """
+        asset_getter = self.get_compute_plan
+        status_failed = models.ComputePlanStatus.failed.value
+        status_canceled = models.ComputePlanStatus.canceled.value
+        statuses_stopped = (
+            models.ComputePlanStatus.done.value,
+            models.ComputePlanStatus.failed.value,
+            models.ComputePlanStatus.canceled.value,
+        )
+
+        return self._wait(
+            key=key,
+            asset_getter=asset_getter,
+            polling_period=polling_period,
+            raises=raises,
+            status_canceled=status_canceled,
+            status_failed=status_failed,
+            statuses_stopped=statuses_stopped,
+            timeout=timeout,
+        )
+
+    @logit
+    def wait_task(
+        self, key: str, *, timeout: Optional[float] = None, polling_period: float = 2.0, raises: bool = True
```

Comment:
Very very minor, but wouldn't it be better to use the imperative for "raise" rather than a descriptive present "raises"? I very well might be wrong (I looked in the code base but couldn't find other examples) but I think of arguments as instructions given to the program.

```suggestion
        self, key: str, *, timeout: Optional[float] = None, polling_period: float = 2.0, raise: bool = True
```

(same for `wait_compute_plan` and `_wait`)

Reply:
`raise` is [a reserved keyword in Python](https://docs.python.org/3/reference/lexical_analysis.html?highlight=reserved#keywords). I also find `raises` not the best solution but I haven't found anything better, if you have any suggestion, please let me know!

Reply:
Lol, good point

Reply:
@ThibaultFy what is your opinion? Do you think `raise_on_failure` better than `raises`(or do you have any other suggestion)?

Reply:
`raises_on_failure` sounds good !